docs/specification.md

```diff
@@ -0,0 +1,1280 @@
+  to all tokens (TODO: discuss).
+
+- Sender must be minter.
+
+- Sender and operation receiver must be whitelisted.
+
+- Contract must not be paused.
+
+**approveOperatorUpdate (TODO)**
+
+Types
+```
+tokenId
+
+operatorTokens =
+  | AllTokens
+  | SomeTokens (set tokenId)
+
+operatorParam =
+  ( address        :owner
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+approveOperatorUpdate = operatorParam
+```
+
+Parameter (in Michelson):
+```
+(pair %approveOperatorUpdate
+  (pair %operator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+    ))
+  (contract %callback
+    (pair %operator
+      (address %owner)
+      (pair
+        (address %operator)
+        (or %tokens
+          (unit %allTokens)
+          (set %someTokens nat)
+        )
+      )
+    )
+  )
+)
+```
+
+- Enables `operator` to operate on tokens held by `owner`.
+
+- `owner` and `operator` must be whitelisted.
+
+- Contract must not be paused.
+
+**isOperator**
+
+Types
+```
+tokenId = nat
+
+operatorTokens =
+  | AllTokens
+  | SomeTokens (set tokenId)
+
+operatorParam =
+  ( address        :owner
+  , address        :operator
+  , operatorTokens :tokens
+  )
+
+isOperatorResponse =
+  ( operatorParam :operator
+  , bool          :isOperator
+  )
+
+isOperatorParam =
+  ( operatorParam               :operator
+  , contract isOperatorResponse :callback
+  )
+
+isOperator = isOperatorParam
+```
+
+Parameter (in Michelson):
+```
+(pair %isOperator
+  (pair %operator
+    (address %owner)
+    (pair
+      (address %operator)
+      (or %tokens
+        (unit %allTokens)
+        (set %someTokens nat)
+      )
+  ))
+  (contract %callback
+    (pair
+      (pair %operator
+        (address %owner)
+        (pair
+          (address %operator)
+          (or %tokens
+            (unit %allTokens)
+            (set %someTokens nat)
+          )
+      ))
+      (bool %isOperator)
+    )
+  )
+)
+```
+
+- Inspect if an address is an operator for the specified owner and    token types.
+
+- If the adddress is not an operator for at least one
+  requested token type, then the result is `false`.
+
+- It's possible to make this query for some specific tokens, or to
+  all tokens.
+
+**Managed Ledger Functions (TBD)**
+----------------------------
+
+TODO: Maybe it's better to omit these mentions since all that left from
+ManagedLedger is their semantics?
+
+Functions for the TZUSDC token implementation found in Tezos [*ManagedLedger*](https://gitlab.com/tzip/tzip/blob/ae2f1e7ebb3454d811a2bea3cd0698b0e64ccea5/proposals/tzip-7/ManagedLedger.md)
+
+**pause**
+
+Types
+```
+pause = (address :sender)
+```
+
+Parameter (in Michelson)
+```
+(pair %pause
+  (address %sender)
+  unit
+)
+```
+
+- Pauses transferring, burning, minting and allowance operations so
+  that they cannot be performed. All other operations remain
+  unaffected.
```

Comment:
And how else this can be implemented then? If you're referring to a `map (address, nat)` so that represents allowance by default. If you mean `setAllowance`-like entrypoints then I think that it's better to be paused as well as other mentioned operations, but I'm not sure.